pkg/generators/builder/builder.go

```diff
@@ -139,33 +189,73 @@ func (b *BuilderPatternGenerator) doesTypeNeedGeneration(t *types.Type) bool {
 	return v
 }
 
+func (b *BuilderPatternGenerator) getTypeEnabledForGeneration(t *types.Type) *types.Type {
```

Comment:
This cloud probably use a doc string. My understanding is that if the immediate parent of a member, embedded or otherwise, was generated than the member gets a generator. Is this a recursion stop?

Reply:
Oh woops yes this needs some explanation.
Take for example [Deployment](https://github.com/10gen/kanopy/blob/master/pkg/builder/k8s/deployment.go#L11), which embeds k8s.io/api [Deployment](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/apps/v1/types.go#L329).
It contains DeploymentSpec and DeploymentStatus.

For codegen, we don't want to generate setters for members are not explicitly enabled. Meaning if we want a setter for DeploymentSpec, we'll need to add this:
```
// +kanopy:builder=true
type DeploymentSpec struct {
	appsv1.DeploymentSpec
}
```

So in this code, Filter function on L243 gets run first and checks if each type is enabled for generation. There the map typesToGenerate gets populated with mapping of `DeploymentSpec` -> our wrapper DeploymentSpec.
In this lookup function, it just checks to see if this type is enabled for generation by checking if entry exists in map. Our wrapper type needs to be returned because that will be used as the input argument for the setter function
```
func (o *Deployment) WithDeploymentSpec(in *DeploymentSpec) *Deployment {.  <- input is our kanopy DeploymentSpec
    if in != nil {
        o.Deployment.Spec = in.DeploymentSpec
    }
    return o
}
```

Reply:
doc string added

Reply:
Seems like this should split into two functions to make it more clear the intent then.  

One function `isTypeEnabled`  and `getWrapperType` (naming can adjust).   

I think it is important that the code is clear where the function is used, and the docstring doesn't help where the function is used.

A good rule to operate by, if adding a comment / documentation especially on an un-exported / private function its either doing more than one thing, not named clearly, or both.

Reply:
Split to `isTypeEnabled` and `getWrapperType`. I can't think of a better name for the latter either